Sources/Sentry/SentryStacktraceBuilder.m

```diff
@@ -56,6 +59,41 @@ - (SentryStacktrace *)retrieveStacktraceFromCursor:(SentryCrashStackCursor)stack
     return stacktrace;
 }
 
+- (SentryStacktrace *)retrieveStacktraceFromCursor:(SentryCrashStackCursor)stackCursor
+{
+    return [self retrieveStackTraceFromCursor:&stackCursor
+                        withFrameSymbolicator:^SentryFrame *(SentryCrashStackCursor *cursor) {
+                            if (stackCursor.symbolicate(cursor)) {
+                                return [self.crashStackEntryMapper mapStackEntryWithCursor:*cursor];
+                            }
+                            return nil;
+                        }];
+}
+
+- (SentryStacktrace *)retrieveStacktraceFromCursorNatively:(SentryCrashStackCursor)stackCursor
+{
+    /**
+     * Different from `retrieveStacktraceFromCursor`, this method uses `dladdr` to retrieve
+     * information from the stack cursor, which is much faster and thread safe but is not async
+     * safe, that's why this method cannot be used during crashes.
+     */
```

Comment:
It looks like `retrieveStacktraceFromCursor` is not used during crashes. Maybe we could change it to use `dladdr`, remove this new method, and keep only one method that is faster, because 'retrieveStacktraceFromCursor' run for every event.

Reply:
Yes, please let's do that. That's an excellent idea, but we need to validate this properly.